internal/pkgsuggest/suggestors/interface.go

```diff
@@ -0,0 +1,6 @@
+package suggestors
+
+type Suggestor interface {
+	IsRelevant(srcDir string) bool
+	Packages(srcDir string) []string
```

Comment:
I know this was modeled after planners, but we should take advantage and improve if we can. Specifically, passing srcDir on every function call (even when not needed) makes code a bit messier. An alternative is to add `srcDir` as a field to suggesters. For example:

```golang
package nginx

type Suggestor struct{
  SrcDir string
}
```

Reply:
I did this but had to pass srcDir to the structs for each suggestor/recommender at initialization. So it looks like this:
```go
func getRecommenders(srcDir string) []recommenders.Recommender {

	return []recommenders.Recommender{
		&dotnet.Recommender{SrcDir: srcDir},
		&golang.Recommender{SrcDir: srcDir},
		&haskell.Recommender{SrcDir: srcDir},
		&java.Recommender{SrcDir: srcDir},
		&javascript.Recommender{SrcDir: srcDir},
		&nginx.Recommender{SrcDir: srcDir},
		&python.RecommenderPip{SrcDir: srcDir},
		&python.RecommenderPoetry{SrcDir: srcDir},
		&ruby.Recommender{SrcDir: srcDir},
		&rust.Recommender{SrcDir: srcDir},
		&zig.Recommender{SrcDir: srcDir},
	}
}
```

Reply:
This feels a bit better but agree not ideal (better because it replaces 2 params with a single field).

BTW, when initializing structs, if you pass the correct types in the correct order, you don't have to specify the field, this allows for less repetition:

```golang
&zig.Recommender{srcDir},
```

Another potentially overengineered solution:


```golang
// base recommender
type Recommender struct {
  srcDir string
}

func (h *Recommender) SetSourceDir(d string) { d.hasSourceDir = d }

type SourceDirSetter interface {
	SetSourceDir(d string)
}
```

You can embed `Recommender` in each recommender struct. And then do:

```golang 
for rec, _ := range recs {
  if r, ok := rec.(SourceDirSetter); ok {
    r.SetSourceDir(srcDir)
  }
}
```

Reply:
I thought of putting a setter but it seemed like it will unnecessarily complicate the logic. I'll do the shorthand {srcDir} at initialization. Even though it is repetitive, it's easier to read and understand later on.

Reply:
Go linter doesn't like the unkeyed fields in struct initialization. Guess I'll just keep it as is 😄